enhancements/vmaas/README.md

```diff
@@ -0,0 +1,297 @@
+        "vm_memory": {
+          "value": "4Gi"
+        }
+      }
+    }
+  }
+}
+```
+
+Once the virtual machine is created, tenants are able to review its current state:
+
+```json
+{
+  "@type": "type.googleapis.com/fulfillment.v1.VirtualMachine",
+  "id": "fecb9b9e-07ac-4d56-8b48-9d50aab71677",
+  "metadata": {
+    "creation_timestamp": "2025-09-17T08:14:17.569076Z",
+    "creators": [
+      "guest"
+    ]
+  },
+  "spec": {
+    "template": "ocp_virt_vm",
+    "template_parameters": {
+      "vm_cpu_cores": {
+        "value": 4
+      },
+      "vm_disk_size": {
+        "value": "30Gi"
+      },
+      "vm_memory": {
+        "value": "4Gi"
+      }
+    }
+  },
+  "status": {
+    "conditions": [
+      {
+        "last_transition_time": "2025-09-19T17:32:24.054439350Z",
+        "message": "",
+        "status": "CONDITION_STATUS_FALSE",
+        "type": "VIRTUAL_MACHINE_CONDITION_TYPE_PROGRESSING"
+      },
+      {
+        "last_transition_time": "2025-09-17T08:52:12.652582382Z",
+        "message": "",
+        "status": "CONDITION_STATUS_TRUE",
+        "type": "VIRTUAL_MACHINE_CONDITION_TYPE_READY"
+      },
+      {
+        "status": "CONDITION_STATUS_FALSE",
+        "type": "VIRTUAL_MACHINE_CONDITION_TYPE_FAILED"
+      },
+      {
+        "status": "CONDITION_STATUS_FALSE",
+        "type": "VIRTUAL_MACHINE_CONDITION_TYPE_DEGRADED"
+      }
+    ],
+    "state": "VIRTUAL_MACHINE_STATE_READY",
+    "internalIP": "10.0.0.1",
+    "externalIP": "193.1.2.3"
+  }
+}
+```
+
+`internalIP` is the IP assigned directly to the virtual machine, and `externalIP` is the floating IP pointing to the virtual machine.
+
+#### VirtualMachineTemplate
+
+Virtual machine template are designed as Ansible roles, these roles must follow the follwing definition:
+
+
+
+There will be a peridioc job witch will publish the Ansible roles as virtual machine templates to the Fulfillment Service, so tenants will be able to reference them to create their virtual machines:
+
+```json
+{
+  "object": {
+    "id": "simple_vm",
+    "title": "Simple VM Template",
+    "description": "This template provisions a virtual machine. CPU, memory, and boot volume size is configurable",
+    "parameters": [
+      {
+        "name": "vm_cpu_cores",
+        "default": {
+          "value": 2
+        }
+      },
+      {
+        "name": "vm_memory",
+        "default": {
+          "value": "2Gi"
+        }
+      },
+      {
+        "name": "vm_disk_size",
+        "default": {
+          "value": "20Gi"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+
+#### Virtual machines on HUB cluster
+
+Virtual machines will be created on the HUB cluster that was selected by Fulfillment Service, it was discussed to create a dedicated cluster to handle VM workloads using Cluster-as-a-Service API, but since they are HostedCluster it won't increase the reliability of the solution, as their reliability are tied to the same HUB cluster.
```

Comment:
I am not a fan of creating this on the HUB cluster long term because I don't think we want to put any GPUs in the hub cluster, they should all be in compute clsuters.  I think its fine for the first MVP to put in the Hub cluster.